src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1207 @@
+                        if r != 0:
+                            break
+                        counter = counter + 1
+
+                    # divide exchange polynomial by this maximal power
+                    laurent_polys[i] = laurent_polys[i] / (
+                        self._polynomial_ring(self._names[j]) ** counter)
+
+        # after performing all substitutions, set internal laurent polynomials
+        self._laurent_polys = laurent_polys
+
+    # mutates self at ith variable
+
+    def mutate(self, i, inplace=True):
+        r"""
+        Mutate this seed at the ``i`` th index.
+
+        INPUT:
+
+        - ``i`` -- integer or iterable of integers; the index/indices to mutate
+          ``self`` at, where we index from 0
+
+        - ``inplace`` -- boolean (default: ``True``); whether to mutate the
+          current instance of the seed, or return a new ``LPASeed`` object
+
+        EXAMPLES:
+
+        We mutate a rank-two Laurent phenomenon algebra at the first index::
+
+            sage: var('x1,x2,a0,a2,b0,b1')
+            (x1, x2, a0, a2, b0, b1)
+            sage: S = LPASeed({x1: a0 + a2*x2, x2: b0 + b1*x1}, coefficients=[a0,a2,b0,b1])
+            sage: print(S)
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2*a2 + a0, x1*b1 + b0]
+            sage: S.mutate(0); print(S)
+            A seed with cluster variables [(x2*a2 + a0)/x1, x2] and exchange polynomials [x2*a2 + a0, x1*b0 + a0*b1]
+
+        Mutating at the same index is an involution::
+
+            sage: var('x1,x2,x3')
+            (x1, x2, x3)
+            sage: S = LPASeed({x1: 1 + x2*x3, x2: 1 + x1^2 + x3^2, x3: 1 + x1 + x2})
+            sage: T = S.mutate([2,2], inplace=False)
+            sage: T == S
+            True
+        """
+
+        # input preprocessing
+
+        if not inplace:
+            self = LPASeed(self)
+
+        # is our input data iterable?
+        if hasattr(i, '__iter__'):
+            for index in i:
+                if index not in range(0, self._rank):
+                    raise IndexError('iterable %s contains a nonvalid index %s'
+                                     % (i, index))
+                self.mutate(index)
+            return self
+
+        elif i not in range(0, self._rank):
+            raise IndexError('did not pass in a valid integer index')
+
+        # Mutate cluster variables:
+
+        xi = self._names[i]  # the variable we are mutating at
+
+        # get the new cluster variable by applying exchange relation
+        # need to cast names as polynomials for subs method
+        d = {str(self._names[j]): self._cluster_vars[j] for j in range(
+            self._rank)}
+        exchange_laurent_poly = self._laurent_polys[i].subs(**d)
+        self._cluster_vars[i] = exchange_laurent_poly / self._cluster_vars[i]
+
+        # Mutate exchange polynomials:
+
+        for j in range(0, self._rank):
+            if (j != i and xi in self._exchange_polys[j].variables()):
```

Comment:
```suggestion
            if j != i and xi in self._exchange_polys[j].variables():
```
Python is not C. ;) In other places too.